core/src/main/java/io/grpc/Attributes.java

```diff
@@ -179,4 +179,8 @@ public Attributes build() {
       return result;
     }
   }
+
+  public interface Provider {
```

Comment:
This adds a public API for an minor internal use. Could this be defined in the netty transport, or something else that similarly removes it from the public API?